exonum/src/blockchain/block.rs

```diff
@@ -60,6 +107,49 @@ pub struct Block {
     /// These calls can include transactions, `before_transactions` and/or `after_transactions` hooks
     /// for services.
     pub error_hash: Hash,
+    /// Some additional entries that can be added into the block.
+    pub entries: BlockHeaderEntries,
+}
+
+impl Block {
+    /// Insert new entry to the block header.
+    ///
+    /// # Usage
+    ///
+    /// ```no_run
+    /// use exonum::blockchain::Block;
+    ///
+    /// let mut block = Block::default();
+    ///
+    /// let services = ActiveServices::new();
+    /// block.insert::<ActiveServices>(services);
+    ///
+    /// ```
+    pub fn insert<K: BlockHeaderKey>(&mut self, value: K::Value) {
```

Comment:
`block.insert(..)` sounds like adding a transaction into it. Maybe `add_entry` / `get_entry`?